File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        opened_stack = []
        
        for c in s:
            if c is '(' or c is '{' or c is '[':
                opened_stack.append(c)
            else:
                top = opened_stack.pop() if len(opened_stack) else None
                if not (top == '(' and c == ')' or top == '{' and c == '}'or top == '[' and c == ']'):
                    return False
 
        return True if len(opened_stack) == 0 else False
                
            

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const stack = []
    const parentheses = {
        '(' : ')',
        '[' : ']',
        '{' : '}'
    }

    for (let i = 0; i < s.length; i++) {
        if(parentheses[s[i]]) {
            stack.push(parentheses[s[i]])
        }
        else {
            if(s[i] === stack[stack.length - 1]) {
                stack.pop()
            }
            else return false
        }
    }
    if(stack.length > 0 ) return false

    return true
    
};